internal/controllers/clusterctlconfig_controller.go

```diff
@@ -0,0 +1,147 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/yaml"
+
+	turtlesv1 "github.com/rancher/turtles/api/v1alpha1"
+	"github.com/rancher/turtles/internal/controllers/clusterctl"
+)
+
+// ClusterctlConfigReconciler reconciles a ClusterctlConfig object.
+type ClusterctlConfigReconciler struct {
+	client.Client
+}
+
+// Config is a direct clusterctl config representation.
+type Config struct {
+	Providers turtlesv1.ProviderList `json:"providers"`
+	Images    map[string]ConfigImage `json:"images"`
+}
+
+// ConfigImage is a direct clusterctl representation of image config value.
+type ConfigImage struct {
+	// Repository sets the container registry override to pull images from.
+	Repository string `json:"repository,omitempty"`
+
+	// Tag allows to specify a tag for the images.
+	Tag string `json:"tag,omitempty"`
+}
+
+func configMapMapper(_ context.Context, obj client.Object) []reconcile.Request {
+	if obj.GetName() != turtlesv1.ClusterctlConfigName {
+		return []reconcile.Request{}
+	}
+
+	return []reconcile.Request{{NamespacedName: client.ObjectKeyFromObject(obj)}}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ClusterctlConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, _ controller.Options) error {
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&turtlesv1.ClusterctlConfig{}).
+		Watches(&corev1.ConfigMap{}, handler.EnqueueRequestsFromMapFunc(configMapMapper)).
+		Complete(r)
+	if err != nil {
+		return fmt.Errorf("creating ClusterctlConfigReconciler controller: %w", err)
+	}
+
+	// This needs to be created in the empty state, so that controller can sync the changes
+	// and have the initial resource to receive reconcile request from
+	configMapTemplate := clusterctl.Config()
+	configMapTemplate.Data = nil
+
+	err = r.Client.Patch(ctx, configMapTemplate, client.Apply, []client.PatchOption{
+		client.ForceOwnership,
+		client.FieldOwner("clusterctl-controller"),
+	}...)
+	if err != nil {
+		return fmt.Errorf("creating ClusterctlConfig default ConfigMap: %w", err)
+	}
+
+	return nil
+}
+
+//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlcofigs,verbs=get;list;watch;patch
+//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlcofigs/status,verbs=get;list;watch;patch
+//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlcofigs/finalizers,verbs=get;list;watch;patch;update
```

Comment:
okay, this is the reason:
```suggestion
//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlconfigs,verbs=get;list;watch;patch
//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlconfigs/status,verbs=get;list;watch;patch
//+kubebuilder:rbac:groups=turtles-capi.cattle.io,resources=clusterctlconfigs/finalizers,verbs=get;list;watch;patch;update
```

Reply:
CAPI provider and internal/controllers with exception of etcd snapshot/restore do not use this kubebuilder generation, to keep the RBAC clean and updated on manual basis. This, if enabled will make duplicate roles, which can be done separately from this PR, as it will need a review.

Reply:
Yes, https://github.com/rancher/turtles/pull/742/files#diff-e4dcbd6018dd457ef50ee530496d1b8c45380f28ed8cbf097d9c8d9ce3025883R3381-R3382 is set to correct value, annotations needs to be fixed, will do in the followup